services/src/datasets/external/nfdi/mod.rs

```diff
@@ -504,25 +579,34 @@ impl MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectan
     > {
         let id: DataId = id.clone().into();
 
-        let (_, md) = self.dataset_info(&id).await.map_err(|e| {
+        let aruna_dataset_ids = self.get_dataset_info(&id).await.map_err(|e| {
             geoengine_operators::error::Error::DatasetMetaData {
                 source: Box::new(e),
             }
         })?;
 
-        let object = self.get_single_file_object(&id).await.map_err(|e| {
+        let meta_data = self.get_metadata(&aruna_dataset_ids).await.map_err(|e| {
             geoengine_operators::error::Error::DatasetMetaData {
                 source: Box::new(e),
             }
         })?;
 
-        match &md.data_type {
+        let aruna_data_object = self
+            .get_single_file_object(&aruna_dataset_ids)
+            .await
+            .map_err(|e| geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(e),
+            })?;
+
+        match &meta_data.data_type {
             DataType::SingleRasterFile(info) => {
-                let result_descriptor = Self::create_raster_result_descriptor(md.crs.into(), info);
+                let result_descriptor =
```

Comment:
you can set the ResultDescriptor resolution and bbox from the rasters GeoTransform and with/height (both are in the RasterInfo).
if aruna can only store single raster file datasets, the `time_interval` in the `RasterInfo` is the temporal bound?

Reply:
92da6f78a46dc3666f36e9a15225addf43a6c333

On a closer look, RasterInfo is kinda redundant with RasterResultDescriptor itself. It would probably be wise to consolidate them somehow, but given the deadline, I won't.